src/java.desktop/unix/classes/sun/awt/X11/XWindowPeer.java

```diff
@@ -206,6 +207,39 @@ void preInit(XCreateWindowParams params) {
         ((X11GraphicsDevice)gc.getDevice()).addDisplayChangedListener(this);
     }
 
+    public GraphicsConfiguration getAppropriateGraphicsConfiguration(
+            GraphicsConfiguration gc)
+    {
+        if (graphicsConfig == null || gc == null) {
+            return gc;
+        }
+
+        final X11GraphicsDevice newDev = getSameScreenDevice(gc);
+        final int visualToLookFor = graphicsConfig.getVisual();
+
+        final GraphicsConfiguration[] configurations = newDev.getConfigurations();
+        for (final GraphicsConfiguration config : configurations) {
+            final X11GraphicsConfig x11gc = (X11GraphicsConfig) config;
+            if (visualToLookFor == x11gc.getVisual()) {
+                graphicsConfig = x11gc;
+            }
+        }
+        return graphicsConfig;
+    }
+
+    private X11GraphicsDevice getSameScreenDevice(GraphicsConfiguration gc) {
+        XToolkit.awtLock();
+        try {
+            final int screenNum = ((X11GraphicsDevice) gc.getDevice()).getScreen();
+            return (X11GraphicsDevice) GraphicsEnvironment.
+                    getLocalGraphicsEnvironment().
+                    getScreenDevices()[screenNum];
+        } finally {
+            XToolkit.awtUnlock();
+        }
+    }
```

Comment:
The same code is already in XCanvasPeer.

XWindowPeer is a descendant of XCanvasPeer, there is no need to repeat the code here.

Reply:
Updated.